### PR TITLE
fix: adapter support added

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**
## Description

Added a feature to automate lockbox adapter process while dealing with xERC20 assets in `xcall`.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Related Issue(s)

Fixes: #5154

### CODE REVIEW

File: packages/agents/sdk/src/sdkShared.ts [hunk 1]
```diff
@@ -179,6 +186,63 @@ export class SdkShared {
...
+
+  async hasLockbox(domainId: string, tokenAddress: string, options?: Options): Promise<boolean> {
+    try {
+      let isValidAsset = false;
+      // Checking if given asset is xERC20 and have lockbox
+      const xerc20Registry = await this.getXERC20Registry(domainId, options);
+      const isXERC20 = await xerc20Registry.functions.isXERC20(tokenAddress);
+      if (isXERC20) {
+        const lockbox = await xerc20Registry.functions.getLockbox(tokenAddress);
+        if (lockbox !== "0x0000000000000000000000000000000000000000") {
```
Comment (just-a-node) [inline packages/agents/sdk/src/sdkShared.ts:236]: we can use `constants.AddressZero`

File: packages/agents/sdk/src/sdkBase.ts [hunk 2]
```diff
@@ -228,7 +228,17 @@ export class SdkBase extends SdkShared {
       });
     }
 
-    const connextContractAddress = (await this.getConnext(origin, options)).address;
+    const isValidAsset = await this.hasLockbox(origin, asset, options);
```
Comment (just-a-node) [inline packages/agents/sdk/src/sdkBase.ts:231]: let's name this something more specfic, like `isValidLockboxAsset`

**Review by just-a-node — DISMISSED** [review]

File: packages/agents/sdk/src/sdkShared.ts (hunk 1 above)
Comment (prathmeshkhandelwal1) [reply packages/agents/sdk/src/sdkShared.ts]: remved this

File: packages/agents/sdk/src/sdkBase.ts (hunk 2 above)
Comment (prathmeshkhandelwal1) [reply packages/agents/sdk/src/sdkBase.ts]: removed this whole logic

**Review by just-a-node — APPROVED** [review]

**Review by liu-zhipeng — APPROVED** [review]

**Pull request merged**